zml/buffer.zig

```diff
@@ -51,8 +51,13 @@ pub const Buffer = struct {
     pub const MAX_NUM_SHARDS: u8 = Platform.MAX_NUM_DEVICES;
     pub const Shards = std.BoundedArray(*pjrt.Buffer, MAX_NUM_SHARDS);
 
+    pub const FromOptions = struct {
+        wait: bool = true,
+        memory: ?pjrt.Memory.Kind = null,
```

Comment:
can we replace the optional by the actual default value ? `.device` IIUC

Reply:
I have mixed feelings about it. I don't know what a sane default would be. For now it's true that on platform we support, it's copied to `.device` memory but will that always be the case ?